### PR TITLE
[FIX] sale_product_configurator: set value for no_variant attribute

### PR DESCRIPTION
**Pull request by vchu-odoo**
**[FIX] sale_product_configurator: allow type False for customValue**

Steps to reproduce(locally):
1) Create SO with the product that has a custom value attribute
2) Leave the custom field empty
3) Save SO and open product configurator again
4) Observe TypeError traceback

Reason: customValue is supposed to be string but if value is not set,
it remains False which cause a type error.

After this commit: allow customValue be false

**[FIX] sale_product_configurator: set value for no_variant attribute**

Steps to reproduce:
1) Create product with an attribute that has several values and with an
attribute with 'create_mode = 'no_variant' and one value.
2) Create SO with this product, save
3) Open product configurator again and see the traceback

Reason: selected_attribute_value_id is not set.

After this commit: the attribure value is defined in get_values of
product configurator.

opw-3513685

**[FIX] sale_product_configurator,product: show archived combination**

Steps to reproduce:
1) Create SO with customizable product (example “Customizable Desk”)
2) Let default values in product configurator
3) Save SO
4) Go to product template, and in the Attribute remove all value
options, only leaving the custom value
5) Try to open the product in configurator in the saved SO

Reason:
archived combination is not loaded

After this commit:
When requested combination is archived, load it

opw-3513685

### CODE REVIEW

File: addons/product/models/product_template.py [hunk 1]
```diff
@@ -779,6 +781,9 @@ def _get_attribute_exclusions(self, parent_combination=None, parent_name=None):
         :type parent_combination: recordset `product.template.attribute.value`
         :param parent_name: the name of the parent product combination.
         :type parent_name: str
+        :param combination: The combination of the product, as a
+            `product.template.attribute.value` ids.
+        :type combination: list
```
Comment (Feyensv) [inline addons/product/models/product_template.py:786]:
```suggestion
        :param list combination: The combination of the product, as a
            list of `product.template.attribute.value` ids.
```

File: addons/product/tests/common.py [hunk 2]
```diff
@@ -86,6 +86,16 @@ def setUpClass(cls):
...
         ) = cls.color_attribute.value_ids
 
+        cls.no_variant_attribute = cls.env['product.attribute'].create(
+            {
+                'name': 'No variant',
+            'create_variant': 'no_variant',
+            'value_ids': [
+                Command.create({'name': 'extra'}),
+            ]
+        })
```
Comment (Feyensv) [inline addons/product/tests/common.py:96]: inconsistent indentation ;)

**Review by Feyensv — COMMENTED** [review]
lgtm, little linting/docstring comments :ok_hand:

**Review by Feyensv — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by vchu-odoo** [comment]
@fw-bot r+

**Pull request closed without merging**